### PR TITLE
Fix bug #7049 causing excessive slowness in expm_multiply

### PR DESCRIPTION
**Pull request by dominicelse**
This fixes the performance bug in expm_multiply reported in #7049.

### CODE REVIEW

File: scipy/sparse/linalg/_expm_multiply.py [hunk 1]
```diff
@@ -595,7 +598,8 @@ def _expm_multiply_interval_core_0(A, X, h, mu, m_star, s, q):
     A helper function, for the case q <= s.
     """
     for k in range(q):
-        X[k+1] = _expm_multiply_simple_core(A, X[k], h, mu, m_star, s)
+        s_step = -(-s//q)  # Ceiling division
```
Comment (pv) [inline scipy/sparse/linalg/_expm_multiply.py]: 1) Is this a recipe in Higham's paper?
2) The comment here needs to explain the reasoning why this is correct.
3) The time step scales as `t/q` but doesn't the scaling scale with `1/2^s`, so is the linear scaling here ok? 
4) Should the per-step s be computed instead with fragment_3_1, before calling core_0 around line 578?

**Review by pv — CHANGES_REQUESTED** [review]

**Comment by dominicelse** [comment]
1. What Higham does is simply call _fragment_3_1 again to compute the new values of s and m_star. As I explain in the comment I added, this is equivalent to simply recalculating s by a ceiling division s/q.

2. Done.

3. I don't understand the question here, could you clarify?

4. See above.

**Comment by pv** [comment]
I understood the added comment so that
```
norm_info2 = LazyOperatorNormInfo(h*A, A_1_norm=h*A_1_norm, ell=ell)
m_star2, s2 = _fragment_3_1(norm_info2, n0, tol, ell=ell)
```
should produces the same `m_star` and `s2 == -(-s//q)`. However, this does not appear to be true always, as can be found by adding some asserts to the code.

**Comment by dominicelse** [comment]
Well, fair enough. The problem is that the ceiling division in the definition of C_m(A) [Eq. (3.10)] in Higham is not quite linear in A due to the rounding to the nearest integer, which can result in a slightly different choice of m_*. These differences are unlikely to be very important, but I suppose that to be on the safe side one could call _fragment_3_1 again.

**Comment by dominicelse** [comment]
I changed the code to explicitly call _fragment_3_1 to compute the optimal values of m_star and s, since this is the optimal thing to do and can be done without substantial extra cost.

**Review by pv — APPROVED** [review]

**Comment by pv** [comment]
Thanks, looks good to go, once travis finishes.

**Pull request merged**